File: website/common/boat.go
package common

import (
	"context"
	"fmt"
	"github.com/jackc/pgx/v4/pgxpool"
)

type Boat struct {
	Boat int
	Shell Shell
	Coxswain Athlete
	Seat8 Athlete
	Seat7 Athlete
	Seat6 Athlete
	Seat5 Athlete
	Seat4 Athlete
	Seat3 Athlete
	Seat2 Athlete
	Seat1 Athlete
}

type BoatModel struct {
	DB *pgxpool.Pool
}

func (m BoatModel) All() ([]Boat, error){
	rows, err := db.Query(context.Background(), "select * from boat")

	if err != nil {
		fmt.Errorf("Error selecting boats %s", err)
	}

	defer rows.Close()

	boats := make([]common.Boat, 0)

	for rows.Next() {
		var boat common.Boat

		err = rows.Scan(&boat.Boat, &boat.Shell, &boat.Coxswain, &boat.Seat8, &boat.Seat7, &boat.Seat6, &boat.Seat5, &boat.Seat4, &boat.Seat3, &boat.Seat2, &boat.Seat1)
		if err != nil {
			fmt.Errorf("Error scanning boat %s", err)
		} else {
			boats = append(boats, boat)
		}
	}

	return boats
}

File: server-java/src/main/java/com/roweatrow/server/model/Boat.java
package com.roweatrow.server.model;

import javax.persistence.*;

@Entity
public class Boat {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long boat;

    @ManyToOne
    @JoinColumn(name = "shell")
    private Shell shell;

    @ManyToOne
    @JoinColumn(name = "coxswain")
    private Athlete coxswain;
    @ManyToOne
    @JoinColumn(name = "seat_8")
    private Athlete seat_8;
    @ManyToOne
    @JoinColumn(name = "seat_7")
    private Athlete seat_7;
    @ManyToOne
    @JoinColumn(name = "seat_6")
    private Athlete seat_6;
    @ManyToOne
    @JoinColumn(name = "seat_5")
    private Athlete seat_5;
    @ManyToOne
    @JoinColumn(name = "seat_4")
    private Athlete seat_4;
    @ManyToOne
    @JoinColumn(name = "seat_3")
    private Athlete seat_3;
    @ManyToOne
    @JoinColumn(name = "seat_2")
    private Athlete seat_2;
    @ManyToOne
    @JoinColumn(name = "seat_1")
    private Athlete seat_1;

    public Athlete getSeat_8() {
        return seat_8;
    }
}
